src/Chapter2.hs

```diff
@@ -841,7 +871,14 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate a arr
+  | a < 0 = []
+  | length arr == 0 = []
+  | a >= 0 = drop moda (take (moda + lenarr) (cycle arr))
```

Comment:
If you do `drop` first, then you can _take_ only `lenarr` elements, and the implementation will be a bit faster :+1: